### PR TITLE
Fix freeze_model call in peft

### PR DESCRIPTION
**Pull request by cuichenx**
# What does this PR do ?

freeze_model() must be called because some peft methods may have custom freeze strategies

**Collection**: [Note which collection this PR will affect]

# Changelog 
- Add specific line by line info of high level changes in this PR.

# Usage
* You can potentially add a usage example below

```python
# Add a code snippet demonstrating how to use this 
```

# GitHub Actions CI

The Jenkins CI system has been replaced by GitHub Actions self-hosted runners.

The GitHub Actions CI will run automatically when the "Run CICD" label is added to the PR.
To re-run CI remove and add the label again.
To run CI on an untrusted fork, a NeMo user with write access must first click "Approve and run".

# Before your PR is "Ready for review"
**Pre checks**:
- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?
  
**PR Type**:
- [ ] New Feature
- [ ] Bugfix
- [ ] Documentation

If you haven't finished some of the above items you can still open "Draft" PR.

## Who can review?

Anyone in the community is free to review the PR once the checks have passed. 
[Contributor guidelines](https://github.com/NVIDIA/NeMo/blob/main/CONTRIBUTING.md) contains specific people who can review PRs to various areas.

# Additional Information
* Related to # (issue)

### CODE REVIEW

File: nemo/lightning/pytorch/callbacks/peft.py [hunk 1]
```diff
@@ -122,12 +119,14 @@ def freeze_model(self, model: nn.Module) -> None:
         Returns:
             nn.Module: The transformed model with PEFT applied.
         """
+        if isinstance(model, MegatronParallel) and len(model) > 1:
```
Comment (akoumpa) [inline nemo/lightning/pytorch/callbacks/peft.py:122]: @cuichenx I was thinking it might be a good idea to also update the type of model?

e.g. `Union[nn.Module, MegatronParallel, torch.nn.parallel.distributed.DistributedDataParallel]` , WDYT?

Comment (cuichenx) [reply nemo/lightning/pytorch/callbacks/peft.py:122]: sure, i can do that

File: nemo/lightning/pytorch/callbacks/peft.py [hunk 2]
```diff
@@ -122,12 +119,14 @@ def freeze_model(self, model: nn.Module) -> None:
...
+            for model_chunk in model:
+                model_chunk.freeze()
         if isinstance(model, torch.nn.parallel.distributed.DistributedDataParallel):
             model.module.freeze()
-            model.module.walk(self.transform)
         else:
             model.freeze()
```
[the remark this answers is not in the document]
Comment (cuichenx) [reply nemo/lightning/pytorch/callbacks/peft.py:128]: freeze() actually comes from `FNMixin`, I can add a check for that, does that work?

**Review by akoumpa — APPROVED** [review]
LGTM, thanks.

File: nemo/lightning/pytorch/callbacks/peft.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (JimmyZhang12) [reply nemo/lightning/pytorch/callbacks/peft.py:128]: yes, thank you

**Review by JimmyZhang12 — APPROVED** [review]
